### DISCUSSION TITLE
Directus must have conditions based on system data

### DISCUSSION THREAD

**Discussion by koksikus** [directus/directus#14828] [Feature Requests]
Suppose such a case, I have a field with the status: 
Draft 
Ready to Publish 
Published 
in addition, I have 3 roles and each can read, create, update, delete each record, but: 
Role A (should set every status) 
Role B (should only set ready to publish, publish should not appear) 
Role C (...) 
to give proper conditions in the status field it was enough to enter System.RoleName.Role A is equal to $ CURRENT_USER. 
Then the user in role A sees the publish status and sets it. Otherwise, I do not see the possibility of setting the condition because if I start with user updated, it may be that either Role A updated the record or Role B, so how to assign the Published status only for Role A...

**Comment by rijkvanzanten** [comment]
@u12206050 this sounds awfully close to what we were discussing earlier

**Reply by u12206050** [reply]
Yeh pretty much. Requires both proper setup of field condition and permission validation with something like $CURRENT_ITEM

**Comment by rijkvanzanten** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
